spec/abci/abci++_app_requirements.md

```diff
@@ -188,6 +188,31 @@ Likewise, `ExtendVote` can also be non-deterministic:
 * *w<sup>r</sup><sub>p</sub> = w<sup>r</sup><sub>q</sub> &#8655;
   e<sup>r</sup><sub>p</sub> = e<sup>r</sup><sub>q</sub>*
 
+Unlike all requirements described above,
+which affect the ABCI methods of the [Consensus Connection](#consensus-connection),
+the following requirement is on the [Mempool Connection](#mempool-connection).
+Let *CheckTxCode<sub>tx,p,t</sub>* denote the error code returned by *p*'s Application, via `ReponseCheckTx`,
+to CometBFT's `RequestCheckTx` call occurring at time *t* and having transaction *tx* as parameter.
+Let predicate *OK(CheckTxCode)* denote whether *CheckTxCode* is `SUCCESS`.
+
+
+* Requirement 13 [`CheckTx`, non-oscillation]: For any correct process *p*
+  and any transaction *tx*,
+  (a) either *OK(CheckTxCode<sub>tx,p,t<sub>0</sub></sub>) = OK(CheckTxCode<sub>tx,p,t<sub>1</sub></sub>)*
+  for any two times *t<sub>0</sub>* and *t<sub>1</sub>* in *p*'s execution, or
+  (b) there exists a time *t<sub>stable</sub>* in *p*'s execution
+  such that *&#172; OK(CheckTxCode<sub>tx,p,t</sub>)* for all *t > t<sub>stable</sub>*.
```

Comment:
Where for all times *t <= t<sub>stable</sub>* , the value was *OK(CheckTxCode<sub>tx,p,t</sub>)* ?

Reply:
Good point. Otherwise there will be no stability until `stable` and if `stable` can take very long to arrive, we are back to square one in practice.

Reply:
I would let @otrack and @hvanz chime in here, but I have the impression we cannot require anything from the app before  t<sub>stable</sub>. This condition is just making sure that we don't oscillate forever: which app can easily fulfill with e.g., nonces. But remember that CheckTx cannot offer strong guarantees when just looking at a particular instance. BTW, I'm adding a sentence at @jmalicevic's suggestion that hopefully clarifies this point. I will extend that sentence with (parts of) this explanation.

Reply:
I see. Maybe a better name for the property would be "eventual stability" or "eventual non-oscillation", given that it can oscillate for a while.